mochimochi/classifier/factory/binary_oml_factory.hpp

```diff
@@ -0,0 +1,261 @@
+/**
+ * Implement the Factory Design Pattern to instanciate Online ML algorithm objects.
+ * https://refactoring.guru/design-patterns/factory-method/cpp/example
+ * 
+ * Also implemented BinaryOMLCreatorInterface to give the option to develop the Proxy design pattern:
+ * https://refactoring.guru/design-patterns/proxy/cpp/example#lang-features
+ */
+
+#ifndef MOCHIMOCHI_BINARY_OML_FACTORY_HPP_
+#define MOCHIMOCHI_BINARY_OML_FACTORY_HPP_
+
+#include <string>
+#include <iostream>
+#include <Eigen/Dense>
+
+#include "binary_oml.hpp"
+#include "../binary/adagrad_rda.hpp"
+#include "../binary/adam.hpp"
+#include "../binary/arow.hpp"
+#include "../binary/nherd.hpp"
+#include "../binary/pa.hpp"
+#include "../binary/scw.hpp"
+
+#include "../../utility/load_svmlight_file.hpp"
+
+using namespace std;
+
+
+
+/**
+ * This interface declares common operations for both BinaryOMLCreator and
+ * whatever Proxy class will be im. As long as the client works with RealSubject using this interface,
+ * you'll be able to pass it a proxy instead of a real subject.
+ */
+class BinaryOMLCreatorInterface
```

Comment:
I just don't like this interface name the "Creator" part should go away if you also use it for proxy, it's a bit misleading and you have "Createor" in the creator class and sub-classes anyway.

Reply:
I've seen "Creator", "Builder", or "Factory" used for this pattern. Which one do you prefer? I'm leaning more towards "Factory" since that's the name of the pattern to being with.

Reply:
Unless you are fine with Creator elsewhere and just want it dropped for this particular interface name: `BinaryOMLInterface`

Reply:
OK, I renamed it. BinaryOMLCreatorInterface --> BinaryOMLInterface. You are correct, this was especially confusing with the IS-A relationship with the Proxy class. Ready for review.